### ISSUE TITLE
Defaults not beign set

### ISSUE THREAD

**Issue by ryanjeffries** [stephane-martin/kafka2influxdb#19]
When having a regex topic such as *-influx-metrics, connection to influx works when no mappings are present and default topic is set. As soon as I bring in a mapping I am getting an unable to connect to influx. Non mapped topics are not getting a default conf values populated.

time="2017-05-18T16:30:53-07:00" level=error msg="Error connecting to InfluxDB" error="Failed to connect to InfluxDB: Unsupported protocol scheme: , your address must start with http:// or https://" function=checkDatabases 
time="2017-05-18T16:30:53-07:00" level=error msg="Error happened while consuming" error="Failed to connect to InfluxDB: Unsupported protocol scheme: , your address must start with http:// or https://" 
time="2017-05-18T16:30:53-07:00" level=info msg=Pausing duration=3000 

Seems to be breaking after fetch configuration from Consul KV commit

**Comment by stephane-martin** [comment]
Yes i agree it is misleading. The code assumes that the user sets a last mapping like '*' => 'default'. Perhaps this "last mapping" should be added automatically if the user has not set it. What do you think ?

**Comment by stephane-martin** [comment]
Please try with the referenced fix.

**Comment by ryanjeffries** [comment]
I will check when I get a chance, I think if the user does not set the "*" => 'default', then the topics coming in from kafka should not be mapped to anything and should not send those metrics to any conf

That way white listing can take place

**Comment by ryanjeffries** [comment]
that fixed it. However, the self-metrics is not taking the default and returning time="2017-05-19T08:33:03-07:00" level=error msg="Error writing self-metrics to InfluxDB" error="Post http://influxdb:8086/write?consistency=&db=k2i_metrics&precision=ns&rp=: dial tcp: lookup influxdb: no such host"

**Comment by stephane-martin** [comment]
for now i reverted "self-metrics". first have to get a look at your PR.

**Comment by stephane-martin** [comment]
further fix: 8adb74fd50ab404eec32661c840a08763faf3a4b
